packages/storage/amplify_storage_s3/lib/method_channel_storage_s3.dart

```diff
@@ -32,10 +33,30 @@ var _transferProgressionCallbackMap =
 
 /// An implementation of [AmplifyPlatform] that uses method channels.
 class AmplifyStorageS3MethodChannel extends AmplifyStorageS3 {
-  AmplifyStorageS3MethodChannel() : super.protected();
+  AmplifyStorageS3PrefixResolver? prefixResolver;
+
+  AmplifyStorageS3MethodChannel({this.prefixResolver}) : super.protected();
+
+  Future<dynamic> _methodCallHandler(MethodCall call) async {
+    switch (call.method) {
+      case 'awsS3PluginPrefixResolver':
+        if (prefixResolver == null) {
+          throw StateError("Native calling nonexistent PrefixResolver in Dart");
+        }
+
+        Map<String, dynamic> arguments =
+            (call.arguments as Map).cast<String, dynamic>();
+
+        return PrefixHandler.handlePrefix(arguments, prefixResolver!);
```

Comment:
nit: maybe we can throw exception on null check of `prefixResolver` instead of force unwrap the nullable property

Reply:
Sounds good, updated: 
```

        Map<String, dynamic>? arguments =
        call.arguments is Map<String, dynamic> ? call.arguments : null;
        
        if(arguments == null) throw StateError("Native send invalid argument to PrefixResolver in Dart");

```